docs/topics/whatsnew17.md

```diff
@@ -0,0 +1,1050 @@
+
+```kotlin
+fun backRef() {
+    val regex = "(?<title>\\w+), yes \\k<title>".toRegex()
+    val match = regex.find("Do you copy? Sir, yes Sir!")!!
+    println(match.value) // "Sir, yes Sir"
+    println(match.groups["title"]?.value) // "Sir"
+}
+```
+
+#### Named groups in replacement expressions
+
+Named group references can be used with replacement expressions. Consider
+the [`replace()`](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.text/-regex/replace.html) function that substitutes all
+occurrences of the specified regular expression in the input with a replacement expression, and
+the [`replaceFirst()`](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.text/-regex/replace-first.html) function that
+swaps the first match only.
+
+Occurrences of `${name}` in the replacement string are substituted with the subsequences corresponding to the captured
+groups with the specified name. You can compare replacements in group references by name and index:
+
+```kotlin
+fun dateReplace() {
+    val dateRegex = Regex("(?<dd>\\d{2})-(?<mm>\\d{2})-(?<yyyy>\\d{4})")
+    val input = "Date of birth: 27-04-2022"
+    println(dateRegex.replace(input, "\${yyyy}-\${mm}-\${dd}")) // "Date of birth: 2022-04-27" — by name
+    println(dateRegex.replace(input, "\$3-\$2-\$1")) // "Date of birth: 2022-04-27" — by number
+}
+```
+
+## Gradle
+
+This release introduces new build reports, support for Gradle plugin variants, new statistics in kapt, and a lot more:
+
+* [A new approach to incremental compilation](#a-new-approach-to-incremental-compilation)
+* [New build reports for tracking compiler performance](#build-reports-for-kotlin-compiler-tasks)
+* [Changes to the minimum supported versions of Gradle and the Android Gradle plugin](#bumping-minimum-supported-versions)
+* [Support for Gradle plugin variants](#support-for-gradle-plugin-variants)
+* [Updates in the Kotlin Gradle plugin API](#updates-in-the-kotlin-gradle-plugin-api)
+* [Availability of the sam-with-receiver plugin via the plugins API](#the-sam-with-receiver-plugin-is-available-via-the-plugins-api)
+* [Changes in compile tasks](#changes-in-compile-tasks)
+* [New statistics of generated files by each annotation processor in kapt](#statistics-of-generated-files-by-each-annotation-processor-in-kapt)
+* [Deprecation of the kotlin.compiler.execution.strategy system property](#deprecation-of-the-kotlin-compiler-execution-strategy-system-property)
+* [Removal of deprecated options, methods, and plugins](#removal-of-deprecated-options-methods-and-plugins)
+
+### A new approach to incremental compilation
+
+> The new approach to incremental compilation is [Experimental](components-stability.md). It may be dropped or
+> changed at any time.
+
+> Opt-in is required (see the details below). We encourage you to Uuse it only for evaluation purposes, and. Wwe would
+> appreciate your feedback in [YouTrack](https://youtrack.jetbrains.com/issues/KT).
+
+>
+{type="warning"}
```

Comment:
```suggestion
> The new approach to incremental compilation is [Experimental](components-stability.md). It may be dropped or changed at any time.
> Opt-in is required (see the details below). We encourage you to use it only for evaluation purposes, and we would
> appreciate your feedback in [YouTrack](https://youtrack.jetbrains.com/issues/KT).
>
{type="warning"}
```